### DISCUSSION TITLE
Debugging why links aren't going to the right anchors

### DISCUSSION THREAD

**Discussion by boshek** [quarto-dev/quarto-cli#9007] [Q&A]
### Description

I was hoping someone here might be able to give some pointers on diagnosing a problem. I am not able to create a minimal reprex. I've tried but the behaviour does not replicate itself outside my quarto website. Here is the issue:

Links with a header anchor are not always arriving at the correct place (but some times they are!). So for example if I navigate to this url:

`www.foo.com/page#anchor`

usually the first time (but not always) the page opens, then scrolls down to the `anchor` section. However on subsequent loads in the same tab/browser window the page scrolls to somewhere below the `anchor` anchor. A simple problem but so far I can't quite figure out how to resolve it. Here is what I've tried so far:

### Disable JavaScript in the browser

The webpage that I am working with has tons of Observable as well as R interfaces to things like reactable. However turning off Javascript did not change this behaviour. 

### Using a fenced div
````
::: {#cool-fig}
{{< include child-docs/_ojs-code.qmd >}}
:::
````

### Renaming a header with a dedicated tag:

```
anchor {#foo}
```

### Tried across different browsers
- chrome
- arc
- firefox
- safari

```
quarto --version 1.4.551
macOS 14.3.1 
```

I think this is all I've tried. Any tips would be much appreciated. I _know_ that a reprex is a preferred but I also feel like it is still worth posting here in case anyone has run into this type of thing before.

**Comment by boshek** [comment]
Adding an empty anchor does seem to make this a little more reliable though not 100%. 

`<a name="int-comp"></a>`

**Comment by mcanouil** [comment]
Could you share a small **self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

**Reply by boshek** [reply]
Is this an auto-reply? I've stated twice in my question that I'm not able to make a reprex. I still think that questions can be posed on this forum that don't have complete reproducible examples. I understand that that is desired (just as I have stated in OP) but I think a space _should_ exist to have discussions that maybe can't exactly be reproduced.

**Reply by mcanouil** [reply]
Here you are mentioning a behaviour that I cannot reproduce and never had.
Did you see the same behaviour across different browser?

How exactly do you want anyone to help?

It's not because I (we) ask you to share something actionable that it means don't post at all.
We can only provide helpful and efficient answers if we can see the issue. Otherwise, we would be guessing randomly, which is not a good use of our time or yours.

**Reply by boshek** [reply]
> Did you see the same behaviour across different browser?
Yes I did. I forgot to include that in my op. Good suggestion. 

> How exactly do you want anyone to help?

From my op: "Any tips would be much appreciated. I know that a reprex is a preferred but I also feel like it is still worth posting here in case anyone has run into this type of thing before."

> It's not because I (we) ask you to share something actionable that it means don't post at all.

Honestly @mcanouil it feels this way with you. If I had a reprex I probably would file a bug report. And when I explicitly mention twice in my original post that I _know_ that a reprex is better but I am posting anyways because I am stuck, it feels like you are telling me to not post. I understand from your post that this is not your intention but I am here to say that that is how it is received.

**Reply by mcanouil** [reply]
Thanks for the feedback.

**Comment by cderv** [comment]
Thanks for sharing your question. It is interesting to ask Q&A to get feedback from the community if they can help with diagnosis. Don't worry about not providing reprex - sometimes it is hard to recreate. However, for anyone looking at this, including us, it will be really hard to help without.... Do you have a live website somewhere that shows the behavior you are describing ? Like a webpage we could look with the Browser developer tools ? 

Honestly, I don't remember encountering this, but this does not feel like something Quarto has an impact on, but rather how browser behaves and maybe how they cache browsing experience... Though I see you tried several browsers so it seems consistent. 🤔 

Maybe the content of the website after being loaded is messing up the x-y-z information and so scrolling is off... 

Honestly, I don't really know why this would happen, but most importantly how Quarto could have an impact on this. 

Sorry - that does not help much, but until I can play with a website where I can observe this, this is the guess I can make 🤷‍♂️ 

Hope you'll be able to find what is happening, or that someone else knowledgable on this could help.

**Reply by boshek** [reply]
Thanks for this message. I really appreciate it. If I ever manage to reproduce on a public website I will certainly come back here and post.
